File: anacondacode/tensorflow_environment/old_code/计算方法/homework5_3.py
# -*- coding: utf-8 -*-
"""
Created on Sat Oct 19 19:02:13 2019

@author: LJZ
"""

#最小二乘法

import numpy as np
import matplotlib.pyplot as plt

n=11
x=[20,40,60,80,100]
y=[4.35,7.55,10.40,13.80,16.80]
x=np.array(x)
y=np.array(y)
n=len(x)
plt.scatter(x,y)
X=np.zeros([n,2])
X[:,0]=np.ones(n)
X[:,1]=x
den=np.matmul(X.T,X)
num=np.dot(X.T,y)
theta=np.linalg.solve(den,num)
plt.plot(x,np.dot(X,theta))

File: matlab_code/oldcode/homework5_3.m
W=[20,40,60,80,100];
F=[4.35,7.55,10.40,13.80,16.80];
n=numel(W);
X=[ones(n,1),W',(W.^2)'];
y=F';
theta=(X'*X)\(X'*y);
scatter(W,F);
hold on;plot(W, X*theta,'LineWidth',2);

% x=[0,1,2,3,5];
% y=[1.1,1.9,3.1,3.9,4.9];
% n=numel(x);
% X=[ones(n,1),x',(x.^2)'];
% y=y';
% theta=(X'*X)\(X'*y);
% scatter(x,y);
% hold on;plot(x, X*theta,'LineWidth',2);